go/pkg/reader/reader.go

```diff
@@ -95,3 +99,69 @@ func (fio *fileSeeker) Initialize() error {
 	fio.initialized = true
 	return nil
 }
+
+type compressedSeeker struct {
+	fs   ReadSeeker
+	encd *zstd.Encoder
+	// This keeps the encrypted data
```

Comment:
I guess you mean "compressed data" here instead?

Reply:
Done.